### PR TITLE
Added template for mold.

### PR DESCRIPTION
**Pull request by basicfunc**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - i686

### CODE REVIEW

**Comment by kawaiiamber** [comment]
Commit message and PR message should be `New Package: mold-1.0.2`. Squash the two commits to one - one commit per package. If [this](https://www.reddit.com/r/voidlinux/comments/siek3g/xbpssrc_help/) is you, could you include full error log?

File: srcpkgs/mold/template [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Template file for 'mold'
+pkgname=mold
+version=1.0.2
+revision=1
+archs="i686 x86_64"
```
Comment (kawaiiamber) [inline srcpkgs/mold/template:5]: Why limit the archs? I've built mold locally on an x86_64-musl arch. You can test build local archs with xbps-src.

File: srcpkgs/mold/template [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+build_style=gnu-makefile
+hostmakedepends="cmake pkg-config"
+makedepends="libstdc++-devel openssl-devel zlib-devel"
+short_desc="A Modern Linker."
```
Comment (kawaiiamber) [inline srcpkgs/mold/template:9]: short_desc shouldn't start with an article (a, an, the) and shouldn't end with a period

**Comment by kawaiiamber** [comment]
If the `do_check` is failing (according to the reddit post), you could override `do_check()` to run true. Though, still interested in why it's failing - full error log.
```
do_check() {
    :
}
```

**Comment by kawaiiamber** [comment]
For `AGPL-3.0-only` license, add
```
post_install() {
    vlicense LICENSE
}
```

**Comment by basicfunc** [comment]
Thanks @kawaiiamber. Yes I'm the same guy from reddit. Should I close this PR as [one another](https://github.com/void-linux/void-packages/pull/35337) is open with v1.0.3?

File: srcpkgs/mold/template (hunk 2 above)
Comment (basicfunc) [reply srcpkgs/mold/template:9]: I'll keep it in mind

File: srcpkgs/mold/template (hunk 1 above)
Comment (basicfunc) [reply srcpkgs/mold/template:5]: I forgot it. Sorry

**Comment by kawaiiamber** [comment]
This PR was opened before that one. I mention the other one just so that they're linked.

**Comment by kawaiiamber** [comment]
Try to keep only one commit since there's only one package - one commit per package (see [CONTRIBUTING.md](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md)). Since there's 12 commits at the moment, reabse and squash them `git rebase -i HEAD~12` and choose `squash`. Change initial commit message to `New package: mold-1.0.2`, or 1.0.3 if you've updated. After squashing, you'll need to force push. `git push origin master -f` For furutre commits, just ammend them to the currect one. `git commit --ammend; git push origin master -f`. If you plan on contributing more, I would also recommend making a branch on your fork to merge against that's not the master branch. People here would usually call that branch the package name. e..g `git checkout -b mold`.

**Comment by kawaiiamber** [comment]
Hi, I noticed you added a template for quicklisp. Usually unless it's a dependancy of another package to PR, separate packages are PR'd separately. This is another reason why having different branches is recommended.

**Comment by kawaiiamber** [comment]
[#31706](https://github.com/void-linux/void-packages/pull/31706) seems to add an older version of mold. Not sure on the status of it, though.

**Pull request closed without merging**